File: TcpSocketSystem_Java/src/com/tecnm/sockets/domain/puertos/IServicioRed.java
package com.tecnm.sockets.domain.puertos;

import java.util.function.Consumer;

/**
 * Contrato para el servicio de red TCP en Java.
 */
public interface IServicioRed {

    boolean estaEscuchando();

    int getPuertoAsignado();

    void iniciarServidor(int puerto, Consumer<ISocketConexion> onClienteConectado) throws Exception;

    void detenerServidor();
}

File: TcpSocketSystem_DotNet/src/Domain/Puertos/IServicioRed.cs
namespace TcpSocketSystem.Core.Domain.Puertos;

/// <summary>
/// Puerto para el servicio de red que abstrae el inicio y parada del servidor de sockets TCP.
/// </summary>
public interface IServicioRed
{
    bool EstaEscuchando { get; }
    int PuertoAsignado { get; }

    Task IniciarServidorAsync(
        int puerto,
        Func<ISocketConexion, CancellationToken, Task> onClienteConectado,
        CancellationToken cancellationToken);

    Task DetenerServidorAsync();
}
